src/sourmash/sig/grep.py

```diff
@@ -0,0 +1,129 @@
+    if args.csv:
+        csv_obj = sourmash_args.FileOutputCSV(args.csv)
+        csv_fp = csv_obj.open()
+        CollectionManifest.write_csv_header(csv_fp)
+
+    # start loading!
+    total_rows_examined = 0
+    for filename in args.signatures:
+        idx = sourmash_args.load_file_as_index(filename,
+                                               yield_all_files=args.force)
+
+        idx = idx.select(ksize=args.ksize,
+                         moltype=moltype,
+                         picklist=picklist)
+
+        # get (and maybe generate) the manifest.
+        manifest = idx.manifest
+        if manifest is None:
+            if require_manifest:
+                error(f"ERROR on filename '{filename}'.")
+                error("sig grep requires a manifest by default, but no manifest present.")
+                error("specify --no-require-manifest to dynamically generate one.")
+                sys.exit(-1)
+            else:
+                manifest = sourmash_args.get_manifest(idx,
+                                                      require=False)
+
+        # find all matching rows.
+        sub_manifest = manifest.filter_on_columns(pattern.search,
+                                                  ["name", "filename", "md5"],
+                                                  invert=args.invert_match)
+        total_rows_examined += len(manifest)
+
+        # write out to CSV, if desired.
+        if args.csv:
+            sub_manifest.write_to_csv(csv_fp)
+
+        # just print out number of matches?
+        if args.count:
+            print_results(f"{len(sub_manifest)} matches: {filename}")
+        elif not args.silent:
+            # nope - do output signatures. convert manifest to picklist, apply.
+            sub_picklist = sub_manifest.to_picklist()
+
+            try:
+                idx = idx.select(picklist=sub_picklist)
+            except ValueError:
+                error("** This input collection doesn't support 'grep' with picklists.")
+                error("** EXITING.")
+                error("**")
+                error("** You can use 'sourmash sig cat' with a picklist,")
+                error("** and then pipe the output to 'sourmash sig grep -")
+                sys.exit(-1)
+
+            # save!
+            for ss in idx.signatures():
+                save_sigs.add(ss)
+    # done with the big loop over all indexes!
+
+    if args.silent:
+        pass
+    else:
+        notify(f"loaded {total_rows_examined} total that matched ksize & molecule type")
+
+        if save_sigs:
+            notify(f"extracted {len(save_sigs)} signatures from {len(args.signatures)} file(s)")
+            save_sigs.close()
+        else:
+            error("no matching signatures found!")
+            sys.exit(-1)
+
+    if args.csv:
+        notify(f"wrote manifest containing all matches to CSV file '{args.csv}'")
+        csv_obj.close()
+
+    if picklist:
+        sourmash_args.report_picklist(args, picklist)
```

Comment:
does this need a test for grep + picklist? I see one for when this will fail (LCA)...

Reply:
ooh, nice catch! `test_sig_grep_7_picklist_md5_lca_fail` tests the failure, but there's no test for the successful use of `--picklist`. Will add.

Reply:
added in 76b2b02f

Reply:
looks good!